bigquery/src/main/java/com/google/cloud/hadoop/io/bigquery/DirectBigQueryRecordReader.java

```diff
@@ -142,7 +142,8 @@ public boolean hasNext() {
       try {
         return !in.isEnd();
       } catch (IOException e) {
-        throw new RuntimeException(e);
+        throw new RuntimeException(
+            "Failed to check for more records:\n" + e.getMessage());
```

Comment:
I meant to ask to add both, a message and a cause, because otherwise cause message will be duplicated in re-thrown exception that could be confusing:
```
throw new RuntimeException("Failed to check for more records", e);
```

Reply:
Oops I meant to do ", e" in addition.

A lot of Hadoop suppresses stacktraces and I was concerned if the message was not in the top error (and in a caused by the user will never see it.)